chrome/content/zotero/xpcom/data/search.js

```diff
@@ -528,9 +528,16 @@ Zotero.Search.prototype.hasPostSearchFilter = function() {
  * Run the search and return an array of item ids for results
  *
  * @param {Boolean} [asTempTable=false]
+ * @param {String[]} [parentKeys=[]] The stack of searches currently being executed.
+ * 		For use internally when executing recursive searches.
  * @return {Promise}
  */
-Zotero.Search.prototype.search = Zotero.Promise.coroutine(function* (asTempTable) {
+Zotero.Search.prototype.search = Zotero.Promise.coroutine(function* (asTempTable, parentKeys = []) {
```

Comment:
Not ideal to expose an internal parameter as the second parameter on the main search method. Maybe convert the whole thing to `options`, with fallback for boolean `asTempTable` and a deprecation warning?